app/services/files_service.ts

```diff
@@ -127,15 +199,47 @@ export default class FilesService {
   /**
    * Deletes a file from storage
    *
-   * @param key - File's key
+   * @param diskKey - File's full key from disk
+   * @return - true if delete is successful, false if no file to delete, throws if error occurs
+   * @throws {FileServiceFileDeleteError} There was an issue deleting the file. Check the cause prop for details.
+   */
+  static async deleteFileWithDiskKey(diskKey: string): Promise<boolean> {
+    try {
+      const recordExists = await FileEntry.deleteByKeyAndReturnResult(
+        this.extractFileKeyFromDiskKey(diskKey),
+      );
+      if (recordExists) {
+        await drive.use().delete(diskKey);
+        return true;
+      }
+      return false;
+    } catch (error) {
+      throw new FileServiceFileDeleteError(error as Error);
+    }
+  }
+
+  /**
+   * Deletes a file from storage
+   *
+   * @param key - File's key without extension. If you have the full diskKey, it is better to use the deleteFileWithDiskKey method
+   * @return - true if delete is successful, false if no file to delete, throws if error occurs
    * @throws {FileServiceFileDeleteError} There was an issue deleting the file. Check the cause prop for details.
    */
-  async deleteFile(key: string): Promise<void> {
-    // Delete file from storage
+  static async deleteFileWithKey(key: string): Promise<boolean> {
     try {
-      await drive.use().delete(key);
+      const extension = await FileEntry.getExtensionIfExists(key);
+      if (extension !== undefined) {
+        await FileEntry.deleteByKey(key);
+        await drive.use().delete(this.mapToDiskKey(key, extension));
+        return true;
+      }
+      return false;
     } catch (error) {
       throw new FileServiceFileDeleteError(error as Error);
     }
```

Comment:
since the primary key only covers the `key` column, there should be no difference between fetching/deleting by a bare key and a full key.
also what about a function to delete using a `FileEntry` instance?